SimPEG/utils/io_utils/io_utils_electromagnetics.py

```diff
@@ -1,70 +1,173 @@
+    if dict_headers is not None:
+        out_headers += dict_headers
+        has_dict = True
+        non_location_columns += len(dict_headers)
+    
+    col_indices = []
+    for h in out_headers:
+        col_indices.append(file_headers.index(h))
+    
+    # Load specified columns of data
+    data_array = np.loadtxt(
+        file_name, comments='!', skiprows=1, usecols=col_indices
+    )
+    n_rows = np.shape(data_array)[0]
+    num_location_columns = np.shape(data_array)[1] - non_location_columns
+    
+    # Extract columns for electrode locations
+    if num_location_columns == 12:
+        a_cols = [0, 1, 2]
+        b_cols = [3, 4, 5]
+        m_cols = [6, 7, 8]
+        n_cols = [9, 10, 11]
+    # 2D survey data or surface 3D data
+    elif num_location_columns == 8:
+        a_cols = [0, 1]
+        b_cols = [2, 3]
+        m_cols = [4, 5]
+        n_cols = [6, 7]
+    # 2D surface data
+    else:
+        a_cols = [1]
+        b_cols = [2]
+        m_cols = [3]
+        n_cols = [4]
+    
+    # Extract electrode locations
+    if is_surface_data:
+        dummy_elevation = 9999  # Taller than mount Everest
+        locations_a = np.c_[data_array[:, a_cols], dummy_elevation*np.ones(n_rows)]
+        locations_b = np.c_[data_array[:, b_cols], dummy_elevation*np.ones(n_rows)]
+        locations_m = np.c_[data_array[:, m_cols], dummy_elevation*np.ones(n_rows)]
+        locations_n = np.c_[data_array[:, n_cols], dummy_elevation*np.ones(n_rows)]
+        warnings.warn(
+            "Loaded data are in surface format. Elevations automatically set to 9999 m. "
+            "Use the project_to_discretized_topography method of the survey to project "
+            "electrode locations to the discretized surface."
+        )
+    else:
+        locations_a = data_array[:, a_cols]
+        locations_b = data_array[:, b_cols]
+        locations_m = data_array[:, m_cols]
+        locations_n = data_array[:, n_cols]
+    
+    
+    survey, out_indices = generate_survey_from_abmn_locations(
+        locations_a=locations_a,
+        locations_b=locations_b,
+        locations_m=locations_m,
+        locations_n=locations_n,
+        data_type=data_type,
+        output_sorting=True
+    )
+    
+    data_object = Data(survey)
+    
+    # Sort and organize all data columns
+    if has_data:
+        data_object.dobs = data_array[out_indices, file_headers.index(data_header)]
+    
+    # Sort and organize all data columns
+    if has_uncert:
+        data_object.standard_deviation = data_array[out_indices, file_headers.index(uncertainties_header)]
+    
+    # Sort and organize all data columns
+    if has_dict:
+        out_dict = {}
+        for h in dict_headers:
+            out_dict[h] = data_array[out_indices, file_headers.index(h)]
+        return data_object, out_dict
+    
+    else:
+        return data_object
 
 
-def read_dcipoctree_ubc(file_name, data_type):
+def read_dcip2d_ubc(file_name, data_type, format_type):
```

Comment:
can `format_type` be a keyword argument with a default value? probably `general`

Reply:
Most of the time, people want to read ubc data in surface format. I really wanted to force people to put in the right file format themselves when loading the data.